File: Java/Arrays/SquaresOfSortedArray.java
class Solution {
    public int[] sortedSquares(int[] A) {
        int[] res = new int[A.length];
        int n = A.length ;
        int i =0; int j = n -1; 
        for(int pos = n-1; pos >= 0; pos--){
            if(Math.abs(A[i]) > Math.abs(A[j])){
                res[pos] = A[i] * A[i];
                i++;
            }else{
                res[pos] = A[j] * A[j];
                j--;
            }
        }
      return res;  
        
    }
}

File: python/SquaresOfSortedArray.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        res = [None] * len(nums) 
        i, j = 0, len(nums)-1
        for index in reversed(range(len(nums))):
            left = nums[i]
            right = nums[j]
            
            if abs(left) > abs(right):
                res[index] = left**2
                i += 1
            else:
                res[index] = right**2
                j -=1
                
        return res